### PR TITLE
Add Zeabur to the hosting comparison (#1059)

### PR DESCRIPTION
**Pull request by KnorpelSenf**
Supersedes #1059.

### CODE REVIEW

File: site/docs/hosting/zeabur-deno.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+const bot = new Bot(token);
+
+bot.command("start", (ctx) => ctx.reply("Hello from Deno & grammY!"));
+
+bot.on("message", (ctx) => ctx.reply("How can I help you?"));
+
+bot.start();
+```
+
+> Note: Get your bot token with [@BotFather](https://t.me/BotFather) on Telegram, and set is as an environment variable `TELEGRAM_BOT_TELEGRAM` in Zeabur.
```
Comment (niusia-ua) [inline site/docs/hosting/zeabur-deno.md:57]:
```suggestion
> Note: Get your bot token with [@BotFather](https://t.me/BotFather) on Telegram, and set is as an environment variable `TELEGRAM_BOT_TOKEN` in Zeabur.
```

File: site/docs/hosting/zeabur-nodejs.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+pnpm install grammy
+
+# Install development dependencies.
+pnpm install -D typescript
+
+# Initialize TypeScript.
+npx tsc --init
+```
+
+Then, cd into `src/`, and create a file named `bot.ts`.
```
Comment (niusia-ua) [inline site/docs/hosting/zeabur-nodejs.md:42]:
```suggestion
Then, `cd` into `src/`, and create a file named `bot.ts`.
```

File: site/docs/hosting/zeabur-deno.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+You can just use the template and start writing your bot's code.
+
+## Deploying
+
+### Method 1: Deploy From GitHub in Zeabur's Dashboard
+
+1. Create a repository on GitHub, it can be public or private and push your code to it.
+2. Go to [Zeabur dashboard](https://dash.zeabur.com).
+3. Click on the `New Project` button, and click on the `Deploy New Service` button, choose `GitHub` as the source and select your repository.
+4. Your service will be deployed automatically.
```
Comment (niusia-ua) [inline site/docs/hosting/zeabur-deno.md:83]:
```suggestion
3. Click on the `New Project` button, and click on the `Deploy New Service` button, choose `GitHub` as the source and select your repository.
4. Go to Variables tab to add your environment variables like `BOT_TOKEN`.
5. Your service will be deployed automatically.
```

Why is the point about environment variables missing here? It is present in the Node.js version.

File: site/docs/hosting/zeabur-nodejs.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+You can just use the template and start writing your bot's code.
+
+## Deploying
+
+### Method 1: Deploy From GitHub in Zeabur's Dashboard
+
+1. Create a repository on GitHub, it can be public or private and push your code to it.
+2. Go to [Zeabur dashboard](https://dash.zeabur.com).
+3. Click on the `New Project` button, and click on the `Deploy New Service` button, choose `GitHub` as the source and select your repository.
+4. Go to Variables tab to add your environment variables like `BOT_TOKEN`.
```
Comment (niusia-ua) [inline site/docs/hosting/zeabur-nodejs.md:126]:
```suggestion
4. Go to Variables tab to add your environment variables like `TELEGRAM_BOT_TOKEN`.
```

**Review by niusia-ua — APPROVED** [review]

File: site/docs/uk/hosting/zeabur-deno.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+prev: false
+next: false
+---
+
+# Хостинг: Zeabur (Deno)
+
+[Zeabur](https://zeabur.com) --- це платформа, яка дозволяє легко розгортати full-stack застосунки.
+Zeabur підтримує різні мови програмування та фреймворки, зокрема Deno та grammY.
+
+У цьому посібнику ви дізнаєтеся, як розгорнути своїх ботів grammY за допомогою Deno на [Zeabur](https://zeabur.com/).
```
Comment (LWJerri) [inline site/docs/uk/hosting/zeabur-deno.md:11]:
```suggestion
У цьому посібнику ви дізнаєтеся, як розгорнути своїх ботів grammY за допомогою Deno на [Zeabur](https://zeabur.com).
```

File: site/docs/hosting/zeabur-deno.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+prev: false
+next: false
+---
+
+# Hosting: Zeabur (Deno)
+
+[Zeabur](https://zeabur.com) is a platform that allows you to deploy your full-stack applications with ease.
+It supports various programming languages and frameworks, including Deno and grammY.
+
+This tutorial will guide you how to deploy your grammY bots with Deno to [Zeabur](https://zeabur.com/).
```
Comment (LWJerri) [inline site/docs/hosting/zeabur-deno.md:11]:
```suggestion
This tutorial will guide you how to deploy your grammY bots with Deno to [Zeabur](https://zeabur.com).
```

File: site/docs/hosting/zeabur-deno.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+::: tip Looking for the Node.js Version?
+This tutorial explains how to deploy a Telegram bot to Zeabur using Deno.
+If you're looking for the Node.js version, please check out [this tutorial](./zeabur-nodejs) instead.
+:::
+
+## Prerequisites
+
+To follow along, you need to have [Github](https://github.com) and [Zeabur](https://zeabur.com/) accounts.
```
Comment (LWJerri) [inline site/docs/hosting/zeabur-deno.md:20]:
```suggestion
To follow along, you need to have [Github](https://github.com) and [Zeabur](https://zeabur.com) accounts.
```

File: site/docs/hosting/zeabur-nodejs.md [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+prev: false
+next: false
+---
+
+# Hosting: Zeabur (Node.js)
+
+[Zeabur](https://zeabur.com) is a platform that allows you to deploy your full-stack applications with ease.
+It supports various programming languages and frameworks, including Node.js and grammY.
+
+This tutorial will guide you how to deploy your grammY bots with Node.js to [Zeabur](https://zeabur.com/).
```
Comment (LWJerri) [inline site/docs/hosting/zeabur-nodejs.md:11]:
```suggestion
This tutorial will guide you how to deploy your grammY bots with Node.js to [Zeabur](https://zeabur.com).
```

File: site/docs/uk/hosting/zeabur-deno.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+::: tip Шукаєте версію для Node.js?
+Цей посібник пояснює, як розгорнути бота Telegram на Zeabur за допомогою Deno.
+Якщо ви шукаєте версію для Node.js, перегляньте [цей посібник](./zeabur-nodejs).
+:::
+
+## Передумови
+
+Щоб слідувати посібнику, вам потрібно мати облікові записи [Github](https://github.com) та [Zeabur](https://zeabur.com/).
```
Comment (LWJerri) [inline site/docs/uk/hosting/zeabur-deno.md:20]:
```suggestion
Щоб слідувати посібнику, вам потрібно мати облікові записи [Github](https://github.com) та [Zeabur](https://zeabur.com).
```

File: site/docs/uk/hosting/zeabur-nodejs.md [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+prev: false
+next: false
+---
+
+# Хостинг: Zeabur (Node.js)
+
+[Zeabur](https://zeabur.com) --- це платформа, яка дозволяє легко розгортати full-stack застосунки.
+Zeabur підтримує різні мови програмування та фреймворки, зокрема Node.js та grammY.
+
+У цьому посібнику ви дізнаєтеся, як розгорнути своїх ботів grammY за допомогою Node.js на [Zeabur](https://zeabur.com/).
```
Comment (LWJerri) [inline site/docs/uk/hosting/zeabur-nodejs.md:11]:
```suggestion
У цьому посібнику ви дізнаєтеся, як розгорнути своїх ботів grammY за допомогою Node.js на [Zeabur](https://zeabur.com).
```

File: site/docs/uk/hosting/zeabur-nodejs.md [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+
+::: tip Шукаєте версію для Deno?
+Цей посібник пояснює, як розгорнути бота Telegram на Zeabur за допомогою Node.js.
+Якщо ви шукаєте версію для Deno, перегляньте [цей посібник](./zeabur-deno).
+:::
+
+## Передумови
+
+Щоб слідувати посібнику, вам потрібно мати облікові записи [Github](https://github.com) та [Zeabur](https://zeabur.com/).
```
Comment (LWJerri) [inline site/docs/uk/hosting/zeabur-nodejs.md:20]:
```suggestion
Щоб слідувати посібнику, вам потрібно мати облікові записи [Github](https://github.com) та [Zeabur](https://zeabur.com).
```

File: site/docs/uk/hosting/zeabur-deno.md [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+
+Потім виконайте наступну команду, щоб запустити бота:
+
+```sh
+deno run --allow-net main.ts
+```
+
+Deno автоматично завантажить залежності, згенерує файл lock і запустить вашого бота.
+
+### Спосіб 2: використання шаблону Zeabur
```
Comment (LWJerri) [inline site/docs/uk/hosting/zeabur-deno.md:69]:
```suggestion
### Спосіб 2: Використання шаблону Zeabur
```

File: site/docs/uk/hosting/zeabur-deno.md [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+
+Zeabur вже надає шаблон, який ви можете використовувати.
+Ви можете знайти його [тут](https://github.com/zeabur/deno-telegram-bot-starter).
+
+Ви можете просто скористатися шаблоном і почати писати код свого бота.
+
+## Розгортання
+
+### Спосіб 1: розгортання з GitHub на інформаційній панелі Zeabur
```
Comment (LWJerri) [inline site/docs/uk/hosting/zeabur-deno.md:78]:
```suggestion
### Спосіб 1: Розгортання з GitHub на інформаційній панелі Zeabur
```

File: site/docs/uk/hosting/zeabur-deno.md [hunk 14]
```diff
@@ -0,0 +1,94 @@
...
+
+1. Створіть репозиторій на GitHub (він може бути публічним або приватним) і перемістіть туди свій код.
+2. Перейдіть на [інформаційну панель Zeabur](https://dash.zeabur.com).
+3. Натисніть на кнопку `New Project`, потім на кнопку `Deploy New Service`, виберіть `GitHub` як джерело і виберіть свій репозиторій.
+4. Перейдіть на вкладку `Variables`, щоб додати ваші змінні оточення, як-от `TELEGRAM_BOT_TOKEN`.
+5. Ваш сервіс буде розгорнуто автоматично.
+
+### Спосіб 2: розгортання за допомогою Zeabur CLI
```
Comment (LWJerri) [inline site/docs/uk/hosting/zeabur-deno.md:86]:
```suggestion
### Спосіб 2: Розгортання за допомогою Zeabur CLI
```

File: site/docs/uk/hosting/zeabur-nodejs.md [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+
+### Спосіб 1: створення нового проєкту з нуля
```
Comment (LWJerri) [inline site/docs/uk/hosting/zeabur-nodejs.md:22]:
```suggestion
### Спосіб 1: Створення нового проєкту з нуля
```

File: site/docs/uk/hosting/zeabur-nodejs.md [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+
+```sh
+pnpm start
+```
+
+> Примітка: щоб запустити бота локально, вам потрібно встановити `ts-node` глобально.
+>
+> Ви можете встановити його за допомогою `pnpm install -g ts-node`.
+
+### Спосіб 2: використання шаблону Zeabur
```
Comment (LWJerri) [inline site/docs/uk/hosting/zeabur-nodejs.md:112]:
```suggestion
### Спосіб 2: Використання шаблону Zeabur
```

File: site/docs/uk/hosting/zeabur-nodejs.md [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+
+Zeabur вже надає шаблон, який ви можете використовувати.
+Ви можете знайти його [тут](https://github.com/zeabur/deno-telegram-bot-starter).
+
+Ви можете просто скористатися шаблоном і почати писати код свого бота.
+
+## Розгортання
+
+### Спосіб 1: розгортання з GitHub на інформаційній панелі Zeabur
```
Comment (LWJerri) [inline site/docs/uk/hosting/zeabur-nodejs.md:121]:
```suggestion
### Спосіб 1: Розгортання з GitHub на інформаційній панелі Zeabur
```

**Review by LWJerri — CHANGES_REQUESTED** [review]
not :shipit:

File: site/docs/uk/hosting/zeabur-deno.md [hunk 18]
```diff
@@ -0,0 +1,94 @@
...
+::: tip Шукаєте версію для Node.js?
+Цей посібник пояснює, як розгорнути бота Telegram на Zeabur за допомогою Deno.
+Якщо ви шукаєте версію для Node.js, перегляньте [цей посібник](./zeabur-nodejs).
+:::
+
+## Передумови
+
+Щоб слідувати посібнику, вам потрібно мати облікові записи [Github](https://github.com) та [Zeabur](https://zeabur.com/).
+
+### Спосіб 1: створення нового проєкту з нуля
```
[the remark this answers is not in the document]
Comment (niusia-ua) [reply site/docs/uk/hosting/zeabur-deno.md:22]: I think this style is not inherent in the Ukrainian language.
@deptyped What do you think?

File: site/docs/uk/hosting/zeabur-deno.md [hunk 19]
```diff
@@ -0,0 +1,94 @@
...
+
+```sh
+# Ініціалізуємо проєкт.
+mkdir grammy-bot
+cd grammy-bot
+
+# Створюємо файл main.ts
+touch main.ts
+
+# Створюємо файл deno.json, щоб згенерувати файл lock
```
Comment (deptyped) [inline site/docs/uk/hosting/zeabur-deno.md:36]:
```suggestion
# Створюємо файл deno.json, щоб згенерувати lock-файл
```

File: site/docs/uk/hosting/zeabur-deno.md [hunk 20]
```diff
@@ -0,0 +1,94 @@
...
+>
+> Ви можете ознайомитися з [цим посібником](https://zeabur.com/docs/deploy/variables) щодо налаштування змінних середовища в Zeabur.
+
+Потім виконайте наступну команду, щоб запустити бота:
+
+```sh
+deno run --allow-net main.ts
+```
+
+Deno автоматично завантажить залежності, згенерує файл lock і запустить вашого бота.
```
Comment (deptyped) [inline site/docs/uk/hosting/zeabur-deno.md:67]:
```suggestion
Deno автоматично завантажить залежності, згенерує lock-файл і запустить вашого бота.
```

File: site/docs/uk/hosting/zeabur-deno.md [hunk 21]
```diff
@@ -0,0 +1,94 @@
...
+Zeabur вже надає шаблон, який ви можете використовувати.
+Ви можете знайти його [тут](https://github.com/zeabur/deno-telegram-bot-starter).
+
+Ви можете просто скористатися шаблоном і почати писати код свого бота.
+
+## Розгортання
+
+### Спосіб 1: розгортання з GitHub на інформаційній панелі Zeabur
+
+1. Створіть репозиторій на GitHub (він може бути публічним або приватним) і перемістіть туди свій код.
```
Comment (deptyped) [inline site/docs/uk/hosting/zeabur-deno.md:80]:
```suggestion
1. Створіть репозиторій на GitHub (він може бути публічним або приватним) і завантажте туди свій код.
```

File: site/docs/uk/hosting/zeabur-nodejs.md [hunk 22]
```diff
@@ -0,0 +1,137 @@
...
+```asciiart:no-line-numbers
+.
+├── node_modules/
+├── src/
+│   └── bot.ts
+├── package.json
+├── pnpm-lock.yaml
+```
+
+Далі нам потрібно додати до нашого `package.json` скрипти `start`.
```
Comment (deptyped) [inline site/docs/uk/hosting/zeabur-nodejs.md:80]:
```suggestion
Далі нам потрібно додати до нашого `package.json` скрипт `start`.
```

File: site/docs/uk/hosting/zeabur-nodejs.md [hunk 23]
```diff
@@ -0,0 +1,137 @@
...
+Zeabur вже надає шаблон, який ви можете використовувати.
+Ви можете знайти його [тут](https://github.com/zeabur/deno-telegram-bot-starter).
+
+Ви можете просто скористатися шаблоном і почати писати код свого бота.
+
+## Розгортання
+
+### Спосіб 1: розгортання з GitHub на інформаційній панелі Zeabur
+
+1. Створіть репозиторій на GitHub (він може бути публічним або приватним) і перемістіть туди свій код.
```
Comment (deptyped) [inline site/docs/uk/hosting/zeabur-nodejs.md:123]:
```suggestion
1. Створіть репозиторій на GitHub (він може бути публічним або приватним) і завантажте туди свій код.
```

**Review by deptyped — APPROVED** [review]

File: site/docs/uk/hosting/zeabur-deno.md (hunk 18 above)
[the remark this answers is not in the document]
Comment (deptyped) [reply site/docs/uk/hosting/zeabur-deno.md:22]: @niusia-ua Yes, I agree. It's grammatically correct to use a lowercase letter here

**Review by LWJerri — APPROVED** [review]
:shipit:

File: site/docs/id/hosting/zeabur-deno.md [hunk 24]
```diff
@@ -0,0 +1,95 @@
...
+
+# Hosting: Zeabur (Deno)
+
+[Zeabur](https://zeabur.com) merupakan sebuah platform yang menawarkan layanan untuk men-deploy aplikasi full-stack kamu dengan mudah.
+Ia mendukung berbagai macam bahasa pemrograman dan framework, termasuk diantaranya adalah Deno dan grammY.
+
+Tutorial ini akan memandu kamu untuk men-deploy bot grammY Deno ke [Zeabur](https://zeabur.com).
+
+::: tip Mencari yang Versi Node.js-nya?
+Tutorial ini ditujukan khusus untuk mendeploy bot Telegram yang dibuat menggunakan Deno ke Zeabur.
```
Comment (dzikrl) [inline site/docs/id/hosting/zeabur-deno.md:14]:
```suggestion
Tutorial ini ditujukan khusus untuk men-deploy bot Telegram yang dibuat menggunakan Deno ke Zeabur.
```

File: site/docs/id/hosting/zeabur-deno.md [hunk 25]
```diff
@@ -0,0 +1,95 @@
...
+
+```sh
+deno run --allow-net main.ts
+```
+
+Perintah di atas akan membuat Deno secara otomatis mengunduh dependensi yang telah ditentukan sebelumnya, membuat lock-file, serta menjalankan bot kamu.
+
+### Cara 2: Menggunakan Templat yang Disediakan oleh Zeabur
+
+Zeabut memiliki sebuah templat yang bisa kamu gunakan sebagai acuan untuk membuat bot Telegram Deno.
```
Comment (dzikrl) [inline site/docs/id/hosting/zeabur-deno.md:71]:
```suggestion
Zeabur memiliki sebuah templat yang bisa kamu gunakan sebagai acuan untuk membuat bot Telegram Deno.
```

File: site/docs/id/hosting/zeabur-nodejs.md [hunk 26]
```diff
@@ -0,0 +1,138 @@
...
+
+# Hosting: Zeabur (Node.js)
+
+[Zeabur](https://zeabur.com) merupakan sebuah platform yang menawarkan layanan untuk men-deploy aplikasi full-stack kamu dengan mudah.
+Ia mendukung berbagai macam bahasa pemrograman dan framework, termasuk diantaranya adalah Node.js dan grammY.
+
+Tutorial ini akan memandu kamu untuk men-deploy bot grammY Node.js ke [Zeabur](https://zeabur.com).
+
+::: tip Mencari yang versi Deno-nya?
+Tutorial ini ditujukan khusus untuk mendeploy bot Telegram yang dibuat menggunakan Node.js ke Zeabur.
```
Comment (dzikrl) [inline site/docs/id/hosting/zeabur-nodejs.md:14]:
```suggestion
Tutorial ini ditujukan khusus untuk men-deploy bot Telegram yang dibuat menggunakan Node.js ke Zeabur.
```

File: site/docs/id/hosting/zeabur-nodejs.md [hunk 27]
```diff
@@ -0,0 +1,138 @@
...
+
+Buat permulaan proyeknya, lalu instal dependensi yang dibutuhkan:
+
+```sh
+# Buat permulaan proyek.
+mkdir grammy-bot
+cd grammy-bot
+pnpm init -y
+
+# Install dependensi utama.
```
Comment (dzikrl) [inline site/docs/id/hosting/zeabur-nodejs.md:32]:
```suggestion
# Instal dependensi utama.
```

File: site/docs/id/hosting/zeabur-nodejs.md [hunk 28]
```diff
@@ -0,0 +1,138 @@
...
+pnpm install grammy
+
+# Install dependensi pengembangan.
```
Comment (dzikrl) [inline site/docs/id/hosting/zeabur-nodejs.md:35]:
```suggestion
# Instal dependensi pengembangan.
```

File: site/docs/id/hosting/zeabur-nodejs.md [hunk 29]
```diff
@@ -0,0 +1,138 @@
...
+pnpm start
+```
+
+> Catatan: Kamu perlu menginstal `ts-node` secara global agar bisa menjalankan bot secara lokal.
+>
+> Kamu bisa menginstalnya dengan menjalankan perintah `pnpm install -g ts-node`.
+
+### Cara 2: Menggunakan Templat yang Disediakan oleh Zeabur
+
+Zeabut memiliki sebuah templat untuk bot Telegram grammY yang bisa kamu gunakan sebagai acuan.
```
Comment (dzikrl) [inline site/docs/id/hosting/zeabur-nodejs.md:114]:
```suggestion
Zeabur memiliki sebuah templat untuk bot Telegram grammY yang bisa kamu gunakan sebagai acuan.
```

**Review by dzikrl — CHANGES_REQUESTED** [review]

File: site/docs/id/hosting/zeabur-deno.md [hunk 30]
```diff
@@ -0,0 +1,95 @@
...
+# Hosting: Zeabur (Deno)
+
+[Zeabur](https://zeabur.com) merupakan sebuah platform yang menawarkan layanan untuk men-deploy aplikasi full-stack kamu dengan mudah.
+Ia mendukung berbagai macam bahasa pemrograman dan framework, termasuk diantaranya adalah Deno dan grammY.
+
+Tutorial ini akan memandu kamu untuk men-deploy bot grammY Deno ke [Zeabur](https://zeabur.com).
+
+::: tip Mencari yang Versi Node.js-nya?
+Tutorial ini ditujukan khusus untuk mendeploy bot Telegram yang dibuat menggunakan Deno ke Zeabur.
+Jika kamu mencari yang versi Node.js-nya, silahkan beralih ke [tutorial ini](./zeabur-nodejs).
```
Comment (dzikrl) [inline site/docs/id/hosting/zeabur-deno.md:15]:
```suggestion
Jika kamu mencari yang versi Node.js-nya, silahkan beralih ke [tutorial berikut](./zeabur-nodejs).
```

File: site/docs/id/hosting/zeabur-nodejs.md [hunk 31]
```diff
@@ -0,0 +1,138 @@
...
+├── package.json
+├── pnpm-lock.yaml
+```
+
+Jika sudah sesuai, tambahkan script `start` ke `package.json`.
+Sekarang, struktur `package.json` kamu semestinya serupa dengan ini:
+
+```json
+{
+  "name": "telegram-bot-starter",
```
Comment (dzikrl) [inline site/docs/id/hosting/zeabur-nodejs.md:85]:
```suggestion
  "name": "bot-telegram-grammy",
```

File: site/docs/id/hosting/zeabur-nodejs.md [hunk 32]
```diff
@@ -0,0 +1,138 @@
...
+  "version": "1.0.0",
+  "description": "Percobaan Telegram Bot menggunakan TypeScript dan grammY",
```
Comment (dzikrl) [inline site/docs/id/hosting/zeabur-nodejs.md:87]:
```suggestion
  "description": "Bot Telegram menggunakan TypeScript dan grammY",
```

**Review by dzikrl — CHANGES_REQUESTED** [review]

**Review by dzikrl — APPROVED** [review]

**Review by quadratz — APPROVED** [review]

**Pull request merged**